lib/oplog/oplogEntry.go

```diff
@@ -125,21 +99,29 @@ func (op *oplogEntry) ChangedFields() []string {
 		// { $v: 1, $set: { "a": 10, "b": 20 }, $unset: { "c": true } }
 
 		fields := []string{}
-		for operationKey, operation := range op.Data {
+		elements, err := op.Data.Elements()
+		if err != nil {
+			metricUnprocessableChangedFields.Inc()
+			log.Log.Errorw("Oplog data for non-replacement v1 update failed to unmarshal",
+				"op", op, "error", err)
+			return []string{}
+		}
+		for _, element := range elements {
+			operationKey := element.Key()
 			if operationKey == "$v" {
 				// $v indicates the update document format; it's not a changed key
 				continue
 			}
 
-			operationMap, operationMapOK := operation.(map[string]interface{})
+			operationMap, operationMapOK := element.Value().DocumentOK()
 			if !operationMapOK {
 				metricUnprocessableChangedFields.Inc()
 				log.Log.Errorw("Oplog data for non-replacement v1 update contained a key with a non-map value",
-					"op", op)
+					"op", op, "operationKey", operationKey, "operationMap", operationMap)
```

Comment:
Since `operationMap` is of the `bson.Raw` type, does logging it give a useful representation of the document?

Reply:
Probably not, I'll remove it